packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -1,154 +1,503 @@
+export interface Chunk<ChunkMetadata extends Jsonifiable = Jsonifiable> {
+  /** The content of this chunk. This is what is provided as context to an LLM when the chunk is selected. */
+  content: string;
+
+  /** The name of the document from which this chunk was extracted. */
+  documentName?: string;
+
+  /** Optional additional metadata associated with this chunk. */
+  metadata?: ChunkMetadata;
+}
+
 /**
- * If you need to chunk your documents, start with this chunker. Move to more powerful chunkers only as needed.
+ * An embedding is a function that maps a string to a vector encoding its semantic meaning.
+ * Often this is based on the same function used to transform text for an LLM's transformers.
  */
-export const defaultChunkMany = async <Metadata extends Jsonifiable = Jsonifiable>(
-  docs: Document<Metadata>[],
-  opts?: Parameters<ChunkMany>[1]
-) => (await Promise.all(docs.map((doc) => defaultChunker(doc, opts)))).flat();
+export interface Embedding {
+  embed(text: string): Promise<number[]>;
+  embedBatch(chunks: string[]): Promise<number[][]>;
+}
+
+export class LangChainEmbeddingWrapper implements Embedding {
+  constructor(readonly lcEmbedding: Embeddings) {}
+
+  embed(text: string): Promise<number[]> {
+    return this.lcEmbedding.embedQuery(text);
+  }
+
+  embedBatch(chunks: string[]): Promise<number[][]> {
+    return this.lcEmbedding.embedDocuments(chunks);
+  }
+}
+
+export const defaultEmbedding = new LangChainEmbeddingWrapper(new OpenAIEmbeddings());
 
 /**
- * A cast wrapper to convert an AI.JSX document to a LangChain document.
+ * An embedded chunk is a piece of a document that is ready to be added into a vector space.
  */
-export function toLangChainDoc(doc: Document): LangChainDocument {
-  return doc as LangChainDocument;
+export interface EmbeddedChunk<ChunkMetadata extends Jsonifiable = Jsonifiable> {
+  /** The identifier for this chunk. For insertions, this will be generated if not populated.*/
+  id?: string;
+
+  /** The vector representing this chunk for semantic nearest neighbors. */
+  vector: number[];
+
+  /** The content of this chunk. This is what is provided as context to an LLM when the chunk is selected. */
+  content: string;
+
+  /** The name of the document from which this chunk was extracted. */
+  documentName?: string;
+
+  /** Optional additional metadata associated with this chunk. */
+  metadata?: ChunkMetadata;
+}
+
+/** A chunk along with its score for a particular query. */
+export interface ScoredChunk<ChunkMetadata extends Jsonifiable = Jsonifiable> {
+  chunk: EmbeddedChunk<ChunkMetadata>;
+  score: number;
+}
+
+export interface Corpus<ChunkMetadata extends Jsonifiable = Jsonifiable> {
+  startCrawl: () => Promise<Corpus.Stats>;
+
+  getStats: () => Promise<Corpus.Stats>;
+
+  search: (
+    query: string,
+    params?: { limit?: number; score_threshold?: number }
+  ) => Promise<ScoredChunk<ChunkMetadata>[]>;
+}
+
+export namespace Corpus {
```

Comment:
I think it might cause confusion in either the tooling or the humans to have a `namespace` and an `interface` with the same name.